File: bench/benches/primes.jl
include(joinpath(@__DIR__, "..", "lib", "bench.jl"))

function isprime(n)
    d = 2
    while d * d <= n
        n % d == 0 && return false
        d += 1
    end
    return true
end

function count_primes(lo, hi)
    c = 0
    for n in lo:hi-1
        isprime(n) && (c += 1)
    end
    return c
end

bench("primes", () -> count_primes(2, 30000))

File: bench/benches/primes.java
// count primes in [2, 30000) by trial division. checksum = 3245.
class Main {
    static boolean isPrime(int n) {
        int d = 2;
        while (d * d <= n) {
            if (n % d == 0) return false;
            d += 1;
        }
        return true;
    }

    static long count(int lo, int hi) {
        long c = 0;
        for (int n = lo; n < hi; n++) {
            if (isPrime(n)) c += 1;
        }
        return c;
    }

    public static void main(String[] a) {
        Bench.bench("primes", () -> count(2, 30000));
    }
}
